packages/generate-arg-types/src/types.ts

```diff
@@ -0,0 +1,143 @@
+import { z } from "zod";
+
+// The names of "control" here are trying to match the names of the controls in Storybook
+// https://storybook.js.org/docs/react/essentials/controls
+
+const common = {
+  required: z.boolean(),
+  label: z.string().optional(),
+};
+
+const Number = z.object({
+  control: z.literal("number"),
+  dataType: z.literal("number"),
+  defaultValue: z.number().nullable(),
+  ...common,
+});
+
+const Range = z.object({
+  control: z.literal("range"),
+  dataType: z.literal("number"),
+  defaultValue: z.number().nullable(),
+  ...common,
+});
+
+const Text = z.object({
+  control: z.literal("text"),
+  dataType: z.string(),
+  defaultValue: z.string().nullable(),
+  ...common,
+});
+
+const MultilineText = z.object({
+  control: z.literal("multiline-text"),
+  dataType: z.literal("string"),
+  defaultValue: z.string().nullable(),
+  ...common,
+});
+
+const Color = z.object({
+  control: z.literal("color"),
+  dataType: z.literal("string"),
+  defaultValue: z.string().nullable(),
+  ...common,
+});
+
+const Boolean = z.object({
+  control: z.literal("boolean"),
+  dataType: z.literal("boolean"),
+  defaultValue: z.boolean().nullable(),
+  ...common,
+});
+
+const Radio = z.object({
```

Comment:
I still don't understand why the data structure doesn't match sb's shape straight from their argTypes documentation `{ contact: { control: 'radio', options: ['email', 'phone', 'mail'] }}`
"contact" is a property name, control and options is the shape I am talking about

Reply:
same for all of them

Reply:
But don't they match?

Our:

```
control: z.literal("radio"),
options: z.array(z.string()),
```

Theirs:

```
{ control: 'radio', options: ['email', 'phone', 'mail'] }
```

Reply:
I am talking about the argTypes shape, we sort of make a mixture out of our shape and argTypes shape, but should we mix them all up? It makes it kinda uncear that those are specifically defined by sb, which might be part of the initial confusion about why we even have these names.

It should actually say somewhere that these are argTypes:

```
const argTypes = {
  label: {
    name: 'label',
    type: { name: 'string', required: false },
    defaultValue: 'Hello',
    description: 'demo description',
    table: {
      type: { summary: 'string' },
      defaultValue: { summary: 'Hello' },
    },
    control: {
      type: 'text'
    }
  }
}
```

Reply:
I'd figure this out as part of https://github.com/webstudio-is/webstudio-builder/issues/1036

Reply:
I disagree, because that issue is about full compatibility with the goal of rendering stories. I am just asking to make sure argTypes shape and format is supported as-is.

Reply:
or did I misunderstand that issue? its not about rendering stories-based components?

Reply:
if not its fine, but I thought just changing the shape to argTypes format doesn't require a separate issue and research, its just a shape and semantics

Reply:
Its just confusing seeing your new shapes you just created to be a mixture of argtypes and webstudio specific types. I think you did it without full understanding and now created an issue to fix what you just created.

Reply:
> or did I misunderstand that issue? its not about rendering stories-based components?

Yes

> if not its fine, but I thought just changing the shape to argTypes format doesn't require a separate issue and research, its just a shape and semantics

It does require a research. 

For example: I know that Storybook supports either an array or an object as `options` of a select. You can do `{options: {"Label 1": "value 1", "Label 2": "value 2"}}`. But this isn't mentioned in the docs.

Basically there's no formal definition of `argTypes` that I know of at the moment.

I want to understand Storybook's interfaces and semantics well before implementing it.

Reply:
Here are the types storybook defines

https://github.com/ComponentDriven/csf/blob/next/src/story.ts#L63

Reply:
It's a complicated type a lot to understand. And also seem incomplete. 

Also seems incomplete. For example there's a `table` property mentioned here https://storybook.js.org/docs/react/api/argtypes but in code there's no table, probably comes goes under `[key: string]: any` https://github.com/ComponentDriven/csf/blob/e86f8c6b1046ca6e529f9713913d7f6b818a3c11/src/story.ts#L46

Reply:
that's the basic csf data types though, here are the control types

https://github.com/storybookjs/storybook/blob/next/code/ui/blocks/src/controls/types.ts

Reply:
Refactored to a format similar to `ArgTypes` without trying to get a strict support as discussed in dm